Always delete empty block on delete

remko opened the pull request:
When hitting 'Delete' on an empty block, deletes the current block (instead of merging in the following block). This behavior is consistent with most other word editors, and also consistent with the case where the empty block is followed by a media block (in which case it already deleted the block).

This resolves #190

hellendag commented on src/model/modifier/RichTextEditorUtil.js:
```diff
@@ -195,13 +195,17 @@ var RichTextEditorUtil = {
+      return EditorState.push(editorState, preserveType, 'remove-range');
+    }
+
+    if (blockAfter.getType() !== 'media') {
```
This can be `atomic` now.

remko commented:
Updated the patch to take into account the media->atomic changes.

remko commented:
I have updated the PR after the changes in the deletion behavior around atomic blocks.

I made the deletion only apply to empty blocks followed by non-atomic blocks. It feels like it would be more consistent if it would also apply to empty blocks followed by atomic blocks (there were no tests failing if I removed this condition from the code), but it seems it was decided in #332 to treat atomic blocks differently in this case, so leaving it like this.

hellendag commented on src/model/modifier/RichTextEditorUtil.js:
```diff
@@ -176,7 +176,32 @@ const RichTextEditorUtil = {
+    }
+
+    // Always delete empty blocks followed by non-atomic blocks
+    if (blockAfter.getType() !== "atomic") {
```
FFR: Single quotes. :)

hellendag commented:
@facebook-github-bot import

facebook-github-bot-5 commented:
Thanks for importing. If you are an FB employee go to [Phabricator](https://our.intern.facebook.com/intern/opensource/github/pull_request/1525808661058014/int_phab) to review.

qrohlf commented:
@hellendag it seems like this was never merged - is the only outstanding issue here the double quotes?

spicyj commented:
My impression is that this should be done at a lower level, like in DraftModifier.removeRange. In particular, here is where we choose to keep the preceding block's attributes:

https://github.com/facebook/draft-js/blob/bb6175cb5038acee8abec3c0a696e4eef5b30e01/src/model/transaction/removeRangeFromContentState.js#L53

It seems like the best way to fix this is to thread through a `removalDirection` boolean down to that function from all of its callers and choose whether to use `startBlock` or `endBlock` appropriately given that direction.

Note that RichTextEditorUtil does not currently handle this logic (DraftModifier and/or removeRangeFromContentState do), so I don't think it makes sense to try to work around that low-level "broken" behavior with a change in a higher-level module.

lacker commented:
It seems like this pull request has been abandoned, so I am going to close it. Please feel free to reopen it if you are still interested in working on it!
